# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/firefox/hotfix-sparc.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- firefox-118.0.2/xpcom/reflect/xptcall/md/unix/xptcinvoke_sparc_solaris.cpp.vanilla	2023-10-14 18:44:45.091217433 +0200
+++ firefox-118.0.2/xpcom/reflect/xptcall/md/unix/xptcinvoke_sparc_solaris.cpp	2023-10-14 18:57:05.548236848 +0200
@@ -22,7 +22,7 @@
     uint32_t result = 0;
     for(uint32_t i = 0; i < paramCount; i++, s++)
     {
-        if(s->IsPtrData())
+        if(s->IsIndirect())
         {
             result++;
             continue;
@@ -90,15 +90,15 @@
     for(uint32_t i = 0; i < l_paramCount; i++, l_d++, l_s++)
     {
 	if (regCount < 5) regCount++;
-        if(l_s->IsPtrData())
+        if(l_s->IsIndirect())
         {
             if(l_s->type == nsXPTType::T_JSVAL)
             {
               // On SPARC, we need to pass a pointer to HandleValue
-              *((void**)l_d) = &l_s->ptr;
+              *((void**)l_d) = &l_s->val.p;
             } else
             {
-              *((void**)l_d) = l_s->ptr;
+              *((void**)l_d) = l_s->val.p;
             }
             continue;
         }
